src/live-programming.rst

```diff
@@ -47,3 +47,225 @@ TODO: in music
+
+Another related research is about aggregate programmaing, which refers to program an aggregate of individual devices as a whole. It provides a way that simplifies the design, creation, and maintenance of complex IoT software systems. The main strategies include making implicit interaction between devices, composing geometric constructions, summarizing data and streaming to other regions, and generalizably constructions for space–time computing.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE3.bib
+    :filter: key == "beal2015aggregate"
+
+  The article describes an application of aggregate programming in IOT field that simplifies the design, creation, and maintenance of complex IoT software systems.
+
+Building Architecture
+~~~~~~~~~~~~~~~~~~~~~
+
+In tradition, architectures like buildings and houses are designed to protect people from external threats, which requires the buildings to be solid and hard. However, this feature also makes the architecture hard to fit the dynamic demands of people to the space. One way to add flexibility to architectures is to apply a programmable physical environment, which combines the flexible software and the solid builidings.
+
+One example is `Squama`, a modular visiblity control of walls and windows for programmable physical architecture.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE.bib
+    :filter: key == "rekimoto2012squama"
+
+  The article presents a programmable wall and window, which can change transparency in modular way. It can enable people to configure the mask area on the surface to have control of certain resources in the real world.
+
+The Squama wall and window is grid-like, where the transparency of each grid can be controlled. One of the usage is to hide confidential content in the room from people outside the wall. With the sensors and camera, the system can track people's sight and give a real-world pixelization to the content inside the room. Also, the system can provide other functions based on people's various needs.
+
+There are also applications in security of physical architecture. One example is a programmable wireless environment for physical layer security.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE2.bib
+    :filter: key == "chen2019intelligent"
+
+  The research team developed an intelligent reflecting surface. By adjusting the reflecting coefficient, the surface can send the incident electromagnetic wave to desired direction, which can improve the secrecy of information transport.
+
+Biology
+~~~~~~~
+
+Programmable physical interfaces have also appeared in biology field. One example is the Body Integrated Programmable Joints Interface described in the following paper.
+
+.. container:: bib-item
+
+  .. bibliography:: programmable_PE2.bib
+    :filter: key == "leigh2016biojoint"
+
+  The project aims at building flexible physical interfaces that enable wearable devices to augment human capabilities.
```

Comment:
This is too vague -- please describe this paper more specifically and in more detail.